### PR TITLE
✨ Base64 encode the filename

### PR DESCRIPTION
**Pull request by Glutexo**
A successor to #49.

### CODE REVIEW

File: test/onigumo_test.exs [hunk 1]
```diff
@@ -3,19 +3,18 @@ defmodule OnigumoTest do
   import Mox
 
   @urls [
-    "http://onigumo.local/hello.html",
-    "http://onigumo.local/bye.html"
+    {"http://onigumo.local/hello.html", "aHR0cDovL29uaWd1bW8ubG9jYWwvaGVsbG8uaHRtbA"},
+    {"http://onigumo.local/bye.html", "aHR0cDovL29uaWd1bW8ubG9jYWwvYnllLmh0bWw"}
```
Comment (Glutexo) [inline test/onigumo_test.exs:7]: This is probably not a good solution. We don’t want to test the encoding algorithm here. We trust the `Base` module and can use it in the test too. It makes the tests more complex as well. Using a Struct would be even better, but that would again increase complexity.

File: test/onigumo_test.exs [hunk 2]
```diff
@@ -47,7 +48,9 @@ defmodule OnigumoTest do
 
     input_path_env = Application.get_env(:onigumo, :input_path)
     input_path_tmp = Path.join(tmp_dir, input_path_env)
-    input_file_content = Enum.map(@urls, &(&1 <> "\n")) |> Enum.join()
+    input_file_content = urls()
+    |> Enum.map(&(&1 <> "\n"))
```
Comment (Glutexo) [inline test/onigumo_test.exs:52]: The capture here is a result of an improper merge. Fixed in #74.

File: test/onigumo_test.exs [hunk 3]
```diff
@@ -3,19 +3,18 @@ defmodule OnigumoTest do
...
   setup(:verify_on_exit!)
 
   @tag :tmp_dir
   test("download a single URL", %{tmp_dir: tmp_dir}) do
     expect(HTTPoisonMock, :get!, &get!/1)
 
-    input_url = Enum.at(@urls, 0)
-    output_path = Path.join(tmp_dir, @output_path)
+    input_url = urls() |> Enum.at(0)
+    output_path = Path.join(tmp_dir, "body.html")
```
Comment (Glutexo) [inline test/onigumo_test.exs:17]: Now, the file name is arbitrary and is used only on this very place. Thus switching to a literal.

File: test/onigumo_test.exs (hunk 2 above)
Comment (Glutexo) [reply test/onigumo_test.exs:52]: #74 is now merged.

File: lib/onigumo.ex [hunk 4]
```diff
@@ -52,4 +50,8 @@ defmodule Onigumo do
   defp http_client() do
     Application.get_env(:onigumo, :http_client)
   end
+
+  def filename(url) do
```
Comment (nappex) [inline lib/onigumo.ex:54]:
```suggestion
  def encode_url(url) do
```

**Review by nappex — COMMENTED** [review]
Conflicts resolved, please check it.

Added some proposal improvements...

**Review by nappex — APPROVED** [review]

File: test/onigumo_test.exs [hunk 5]
```diff
@@ -108,4 +104,12 @@ defmodule OnigumoTest do
   defp body(url) do
     "Body from: #{url}\n"
   end
+
+  defp assert_downloaded(url, tmp_dir) do
+    filename = Base.url_encode64(url, padding: false)
+    output_path = Path.join(tmp_dir, filename)
```
Comment (nappex) [inline test/onigumo_test.exs:110]:
```suggestion
    output_path = Path.join(tmp_dir, file_name)
```

File: test/onigumo_test.exs [hunk 6]
```diff
@@ -14,39 +14,22 @@ defmodule OnigumoTest do
...
 
     input_url = Enum.at(@urls, 0)
-    output_path = Path.join(tmp_dir, "body.html")
-    download_result = Onigumo.download_url(
-      input_url, HTTPoisonMock, output_path
-    )
+    download_result = Onigumo.download_url(input_url, tmp_dir)
     assert(download_result == :ok)
 
+    output_path = Path.join(tmp_dir, Onigumo.filename(input_url))
```
Comment (Glutexo) [inline test/onigumo_test.exs]:
```suggestion
    output_filename = Onigumo.filename(input_url)
    output_path = Path.join(tmp_dir, output_filename)
```

File: test/onigumo_test.exs [hunk 7]
```diff
@@ -98,7 +81,8 @@ defmodule OnigumoTest do
   @tag :tmp_dir
   test("write response to file", %{tmp_dir: tmp_dir}) do
     response = "Response!"
-    output_path = Path.join(tmp_dir, @output_path)
+    file_name = "testfile"
```
Comment (Glutexo) [inline test/onigumo_test.exs]:
```suggestion
    file_name = "body.html"
```

```suggestion
    output_filename = "testfile"
```

File: lib/onigumo.ex (hunk 4 above)
Comment (Glutexo) [reply lib/onigumo.ex]:
```suggestion
  def url_to_filename(url) do
```

```suggestion
  def create_filename(url) do
```

**Review by nappex — APPROVED** [review]

**Comment by Glutexo** [comment]
Thanks for merging the conflict and for your suggestions. This was an awesome piece of work!

**Pull request merged**